built_in_tasks/target_capture_task.py

```diff
@@ -1062,3 +1062,332 @@ def sequence_2D(nblocks=100, distance=6.8):
+        if self.target_index == -1:
+
+            # Came from a penalty state
+            pass
+        elif self.target_index + 1 < self.chain_length:
+
+            # Hide the current target if there are more
+            self.targets[self.target_index % 2].hide()
+            self.sync_event('EYE_TARGET_OFF', self.gen_indices[self.target_index])
+
+    def _start_timeout_penalty(self):
+        super()._start_timeout_penalty()
+        self.num_hold_state = 0
+        for target in self.targets_hand:
+            target.hide()
+            target.reset()
+            
+    def _start_hold_penalty(self):
+        super()._start_hold_penalty()
+        self.num_hold_state = 0
+        # Hide targets
+        for target in self.targets_hand:
+            target.hide()
+            target.reset()
+
+    def _start_delay_penalty(self):
+        super()._start_delay_penalty()
+        # Hide targets
+        for target in self.targets_hand:
+            target.hide()
+            target.reset()
+
+    def _start_fixation_penalty(self):
+        self._increment_tries()
+        self.sync_event('FIXATION_PENALTY') 
+
+        # Hide targets
+        for target in self.targets:
+            target.hide()
+            target.reset()
+
+        for target in self.targets_hand:
+            target.hide()
+            target.reset()
+
+    def _end_fixation_penalty(self):
+        self.sync_event('TRIAL_END')
+
+    def _start_reward(self):
+        super()._start_reward()
+        # Hide targets
+        for target in self.targets_hand:
+            target.hide()
+            target.reset()        
+
+    # Generator functions
+    @staticmethod
+    def row_target(nblocks=100, ntargets=3, dx=3., origin=(0,0,0)):
+        '''
+        Generates a sequence of 2D (x and z) targets at a given distance from the origin
+
+        Parameters
+        ----------
+        nblocks : int
+            The number of ntarget pairs in the sequence.
+        ntargets : int
+            The number of equally spaced targets
+        distance : float
+            The distance in cm between the center and peripheral targets.
+        origin : 3-tuple
+            Location of the central targets around which the peripheral targets span
+
+        Returns
+        -------
+        [nblocks*ntargets x 1] array of tuples containing trial indices and [1 x 3] target coordinates
+
+        '''
+        rng = np.random.default_rng()
+        for _ in range(nblocks):
+            order = np.arange(ntargets) + 1 # target indices, starting from 1
+            rng.shuffle(order)
+            x_pos_candidate = [-dx,0,dx]
+            for t in range(ntargets):
+                idx = order[t]
+                x_pos = x_pos_candidate[idx-1]
+                pos = np.array([x_pos,0,0]).T
+                yield [idx], [pos + origin]
+
+    @staticmethod
+    def sac_hand_2d(nblocks=100, ntargets=3, dx=10, origin1=(0,0,-1.0), origin2=(0,0,-5.5), origin3=(0,0,5.5)):
```

Comment:
it doesn't make sense to have more than 1 origin.. i suggest a single origin plus 3 offsets (hold_z, targ_z1, targ_z2, e.g.)

Reply:
or whatever you want. i just wouldn't call them origin1 origin2 origin3